homeassistant/components/idasen_desk/button.py

```diff
@@ -0,0 +1,99 @@
+"""Representation of Idasen Desk buttons."""
+from collections.abc import Callable, Coroutine
+from dataclasses import dataclass
+import logging
+from typing import Any, Final
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_NAME, EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import DeskData, IdasenDeskCoordinator
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class IdasenDeskButtonDescriptionMixin:
+    """Mixin to describe a IdasenDesk button entity."""
+
+    press_action: Callable[
+        [IdasenDeskCoordinator], Callable[[], Coroutine[Any, Any, Any]]
+    ]
+
+
+@dataclass
+class IdasenDeskButtonDescription(
+    ButtonEntityDescription, IdasenDeskButtonDescriptionMixin
+):
+    """Class to describe a IdasenDesk button entity."""
+
+
+BUTTONS: Final = [
+    IdasenDeskButtonDescription(
+        key="connect",
+        name="Connect",
+        device_class=ButtonDeviceClass.RESTART,
+        entity_category=EntityCategory.CONFIG,
+        press_action=lambda coordinator: coordinator.async_connect,
+    ),
+    IdasenDeskButtonDescription(
+        key="disconnect",
+        name="Disconnect",
+        device_class=ButtonDeviceClass.RESTART,
+        entity_category=EntityCategory.CONFIG,
+        press_action=lambda coordinator: coordinator.async_disconnect,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set buttons for device."""
+    data: DeskData = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        IdasenDeskButton(data.address, data.device_info, data.coordinator, button)
+        for button in BUTTONS
+    )
+
+
+class IdasenDeskButton(ButtonEntity):
+    """Defines a IdasenDesk button."""
+
+    entity_description: IdasenDeskButtonDescription
+
+    def __init__(
+        self,
+        address: str,
+        device_info: DeviceInfo,
+        coordinator: IdasenDeskCoordinator,
+        description: IdasenDeskButtonDescription,
+    ) -> None:
+        """Initialize the IdasenDesk button entity."""
+        self.entity_description = description
+
+        self._attr_name = f"{device_info[ATTR_NAME]} {self.entity_description.key}"
```

Comment:
https://developers.home-assistant.io/docs/core/entity/#has_entity_name-true-mandatory-for-new-integrations

You can use has entity name instead